internal/common/docker/images.go

```diff
@@ -0,0 +1,54 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package docker // import "github.com/open-telemetry/opentelemetry-collector-contrib/internal/common/docker"
+
+import (
+	"fmt"
+	"regexp"
+
+	"go.uber.org/zap"
+)
+
+var (
+	extractImageRegexp = regexp.MustCompile(`^(?P<repository>([^/\s]+/)?([^:\s]+))(:(?P<tag>[^@\s]+))?(@sha256:\d+)?$`)
+)
+
+// ParseImageName extracts image repository and tag from a combined image reference
+// e.g. example.com:5000/alpine/alpine:test --> `example.com:5000/alpine/alpine` and `test`
+func ParseImageName(image string) (string, string, error) {
```

Comment:
Should we better return a struct "ContaninerImage { repository, tag }"? Maybe in the future we want to return the sha256 or other things?

Reply:
This could be done when we see that use case creeping up, but currently what I can see there is only a need to parse image strings like they are provided in Kubernetes.

Reply:
Using SHAs is definitely a common use case, especially for disconnected installations.